C# release changes for `cargo release`

jdetter opened the pull request:
# Description of Changes

<!-- Please describe your change, mention any related tickets, and so on here. -->

This adds the new "skeleton" directory that we use for `cargo release`.

# API and ABI breaking changes

<!-- If this is an API or ABI breaking change, please apply the
corresponding GitHub label. -->

None

# Expected complexity level and risk

1 - this is just a release change

<!--
How complicated do you think these changes are? Grade on a scale from 1 to 5,
where 1 is a trivial change, and 5 is a deep-reaching and complex change.

This complexity rating applies not only to the complexity apparent in the diff,
but also to its interactions with existing and future code.

If you answered more than a 2, explain what is complex about the PR,
and what other components it interacts with in potentially concerning ways.  -->

# Testing

<!-- Describe any testing you've done, and any testing you'd like your reviewers to do,
so that you're confident that all the changes work as expected! -->

- [x] We used exactly this for the `1.8.0` and `1.9.0` release and we'll also be using it for the `1.10.0` release.

jdetter committed:
Changes for `cargo release`

jdetter committed:
Improve CI check

jdetter committed:
Ignore meta files

jdetter committed:
Merge branch 'master' into jdetter/csharp-release-changes

jdetter committed:
Update sdks/csharp/release~/spacetimedb.bsatn.runtime/unversioned/.gitignore

Co-authored-by: Zeke Foppa <196249+bfops@users.noreply.github.com>
Signed-off-by: John Detter <4099508+jdetter@users.noreply.github.com>

bfops reviewed (changes requested):
Can you also remove the files from sdks/csharp/packages? They are stale since we're no longer updating them, and then I believe many of these files will show up as moved.

Also, I think we should probably `gitignore` them, so that devs iterating locally on `sdks/csharp` will not constantly have dirty working directories.

jdetter committed:
Update sdks/csharp/release~/.gitignore

Co-authored-by: Zeke Foppa <196249+bfops@users.noreply.github.com>
Signed-off-by: John Detter <4099508+jdetter@users.noreply.github.com>

jdetter committed:
Remove old package

jdetter committed:
Merge branch 'jdetter/csharp-release-changes' of ssh://github.com/clockworklabs/SpacetimeDB into jdetter/csharp-release-changes

jdetter commented:
@bfops I removed everything under `sdks/csharp/packages/*` but I left the `packages` directory itself so that the `.meta` file can be maintained there. I could move the `packages` directory under `release~` so that it's `release~/packages/spacetimedb.bsatn...` what do you think?

jdetter committed:
Moved packages under release~

jdetter committed:
Merge branch 'master' into jdetter/csharp-release-changes

jdetter committed:
Merge tag 'v1.11.0' into jdetter/csharp-release-changes

jdetter committed:
Move gitignore file

jdetter committed:
Remove unused gitignore file

jdetter committed:
Update typescript package size limits (#3854)

# Description of Changes

<!-- Please describe your change, mention any related tickets, and so on
here. -->

Updated package sizes from a failed release dry-run:
https://github.com/clockworklabs/SpacetimeDBPrivate/actions/runs/20045893743/job/57491806595

# API and ABI breaking changes

<!-- If this is an API or ABI breaking change, please apply the
corresponding GitHub label. -->

None

# Expected complexity level and risk

1 - this is just a release fix

<!--
How complicated do you think these changes are? Grade on a scale from 1
to 5,
where 1 is a trivial change, and 5 is a deep-reaching and complex
change.

This complexity rating applies not only to the complexity apparent in
the diff,
but also to its interactions with existing and future code.

If you answered more than a 2, explain what is complex about the PR,
and what other components it interacts with in potentially concerning
ways. -->

# Testing

<!-- Describe any testing you've done, and any testing you'd like your
reviewers to do,
so that you're confident that all the changes work as expected! -->

I have not tested this, it is a trivial change.

bfops reviewed (approved):
This LGTM, minus my two comments

bfops reviewed (dismissed):
Actually, one other question - have you thought about the flow for devs making modifications to the Unity SDK? Does it still work smoothly?

My assumption would be that they end up with a version directly that keeps wanting to be added to `git`, so maybe we should `gitignore` any versioned directories, and temporarily remove that `.gitignore` file when we're doing the release?

I do want to make sure we don't make a bad flow for devs doing Unity SDK things, even if we don't do all of it in this PR. We may want an extra README and/or script to help facilitate the meta file copying.

jdetter committed:
Release v1.11.1

jdetter committed:
Merge tag 'v1.11.1' into jdetter/csharp-release-changes

jdetter committed:
Update .gitignore

Signed-off-by: John Detter <4099508+jdetter@users.noreply.github.com>

bfops reviewed (approved):
This LGTM as long as we build that utility command for doing the DLL dance soon (although I'm not the one who will be sad if it gets delayed)
